stl/inc/tuple

```diff
@@ -600,7 +600,9 @@ public:
 
     template <class... _Other>
     constexpr bool _Equals(const tuple<_Other...>& _Right) const {
-        return _Myfirst._Val == _Right._Myfirst._Val && _Mybase::_Equals(_Right._Get_rest());
+        // GH-3131, operator== is only required to return a type that is convertible to bool

```

Comment:
We conventionally don't track history in the source itself; the comments would quickly get out of hand. We _do_ use comments to annotate when our implementation deviates from the standard, but this isn't a deviation.